jupyter_server_fileid/manager.py

```diff
@@ -699,39 +740,33 @@ def get_path(self, id):
         prior to calling `get_path()`.
         """
         # optimistic approach: first check to see if path was not yet moved
-        row = self.con.execute("SELECT path, ino FROM Files WHERE id = ?", (id,)).fetchone()
-
-        # if file ID does not exist, return None
-        if not row:
-            return None
+        for retry in [True, False]:
+            row = self.con.execute(
+                "SELECT path, ino, crtime, mtime FROM Files WHERE id = ?", (id,)
+            ).fetchone()
 
-        path, ino = row
-        stat_info = self._stat(path)
-
-        if stat_info and ino == stat_info.ino and self._check_timestamps(stat_info):
-            # if file already exists at path and the ino and timestamps match,
-            # then return the correct path immediately (best case)
-            return self._from_normalized_path(path)
-
-        # otherwise, try again after calling _sync_all() to sync the Files table to the file tree
-        self._sync_all()
-        row = self.con.execute("SELECT path, ino FROM Files WHERE id = ?", (id,)).fetchone()
-        # file ID already guaranteed to exist from previous check
-        path, ino = row
+            # if file ID does not exist, return None
+            if not row:
+                return None
 
-        # if file no longer exists at path, return None
-        stat_info = self._stat(path)
-        if stat_info is None:
-            return None
+            path, ino, crtime, mtime = row
+            stat_info = self._stat(path)
 
-        # if inode numbers or timestamps of the file and record don't match,
-        # return None
-        if ino != stat_info.ino or not self._check_timestamps(stat_info):
+            if (
+                stat_info
+                and ino == stat_info.ino
+                and self._check_timestamps(stat_info, crtime, mtime)
+            ):
+                # if file already exists at path and the ino and timestamps match,
+                # then return the correct path immediately (best case)
+                return self._from_normalized_path(path)
+
+            # otherwise, try again after calling _sync_all() to sync the Files table to the file tree
+            if retry:
+                self._sync_all()
+        else:
             return None
```

Comment:
I don't think the `else` block is necessary given that we're not breaking from the loop.

Reply:
True.  I'll remove it.